mirar/pipelines/winter/generator.py

```diff
@@ -536,46 +543,52 @@ def winter_reference_generator(image: Image):
+                accepted_values=[fieldid, subdetid],
+            )
+
+            db_results = select_from_table(
+                db_constraints=constraints,
+                sql_table=RefStack.sql_model,
+                output_columns=["savepath"],
+            )
 
-        db_results = select_from_table(
-            db_constraints=constraints,
-            sql_table=RefStack.sql_model,
-            output_columns=["savepath"],
+            if len(db_results) > 0:
+                savepath = db_results["savepath"].iloc[0]
+                if os.path.exists(savepath):
+                    logger.debug(f"Found reference image in database: {savepath}")
+                    return RefFromPath(path=savepath, filter_name=filtername)
+
+        skip_online_query = filtername == "H"
+        ukirt_query = UKIRTOnlineQuery(
+            num_query_points=16,
+            filter_name=filtername,
+            use_db_for_component_queries=True,
+            components_db_table=RefComponent,
+            query_db_table=RefQuery,
+            skip_online_query=skip_online_query,
+            component_image_subdir="winter/references/components",
+        )
+        return WFCAMStackedRef(
+            filter_name=filtername,
+            wfcam_query=ukirt_query,
+            image_resampler_generator=winter_wfau_component_image_stacker,
+            write_stacked_image=cache_ref_stack,
+            write_stack_sub_dir="winter/references/ref_stacks",
+            write_stack_to_db=cache_ref_stack,
+            stacks_db_table=RefStack,
+            component_image_sub_dir="components",
+            references_base_subdir_name="winter/references",
+            stack_image_annotator=winter_reference_stack_annotator,
         )
 
-        if len(db_results) > 0:
-            savepath = db_results["savepath"].iloc[0]
-            if os.path.exists(savepath):
-                logger.debug(f"Found reference image in database: {savepath}")
-                return RefFromPath(path=savepath, filter_name=filtername)
-
-    ukirt_query = UKIRTOnlineQuery(
-        num_query_points=9,
-        filter_name=filtername,
-        use_db_for_component_queries=True,
-        components_db_table=RefComponent,
-        query_db_table=RefQuery,
-        skip_online_query=False,
-        component_image_subdir="winter/references/components",
-    )
-    return WFCAMStackedRef(
-        filter_name=filtername,
-        wfcam_query=ukirt_query,
-        image_resampler_generator=winter_wfau_component_image_stacker,
-        write_stacked_image=cache_ref_stack,
-        write_stack_sub_dir="winter/references/ref_stacks",
-        write_stack_to_db=cache_ref_stack,
-        stacks_db_table=RefStack,
-        component_image_sub_dir="components",
-        references_base_subdir_name="winter/references",
-        stack_image_annotator=winter_reference_stack_annotator,
-    )
+    # Use PS1 references for Y-band
+    logger.debug("Will query reference image from PS1")
```

Comment:
please add an assert or other check here, to be sure it's really y. If they somehow get nonsense filter values assigned, it'll be easier to debug if we catch it here.